src/services/kadenaService.js

```diff
@@ -1,19 +1,32 @@
+/* eslint-disable no-loop-func */
 /* eslint-disable no-await-in-loop */
 const axios = require('axios');
+const rateLimit = require('axios-rate-limit');
 const https = require('https');
 const config = require('config');
 const LRU = require('lru-cache');
 const log = require('../lib/log');
 const serviceHelper = require('./serviceHelper');
 const fluxService = require('./fluxService');
 
+const http = rateLimit(axios.create(), { maxRequests: 10, perMilliseconds: 1000 });
+
 let db = null;
 const kadenaNodesCollection = config.database.kadena.collections.nodes;
 const completedRoundsCollection = config.database.kadena.collections.completedRounds;
 
 let outdatedKDANodes = [];
 let currentNodes = [];
 let allNodes = [];
+let allKDANodes = [];
```

Comment:
this looks like to many variables now and is quite complicated. Would be good to simplify it